### ISSUE TITLE
Optimize Array.Reverse for Write Register

### ISSUE THREAD

**Issue by Touseefelahi** [Stira-sa/GigeVision#7]
As discussed that GVCP is asking for Register Address in Big Endiance so we should convert the address in big endiance while reading the XML file.

#### Simple Test Criteria would be as follows:

1. Get Register Address from the XML file 
2. Decode it to Long as Big Endian 
3. Use BitConverter for the converted Long value
4. Compare the Raw Converted bytes with the Address from XML (They should be same)

**Comment by FaisalFagihi** [comment]
Result:

Raw: 0x12345678
Converted: 0x7856341200000000

**Comment by Touseefelahi** [comment]
> Result:
> 
> Raw: 0x12345678 Converted: 0x7856341200000000

Are they equal to you?

**Comment by Touseefelahi** [comment]
I mean the in the fourth step, the result is equal?

**Comment by FaisalFagihi** [comment]
> > Result:
> > Raw: 0x12345678 Converted: 0x7856341200000000
> 
> Are they equal to you?

Obviously they are not.

**Comment by Touseefelahi** [comment]
@FaisalFagihi is there any problem with the test procedure?

**Comment by FaisalFagihi** [comment]
@Touseefelahi I have no Idea about the excepted result. I just followed the procedure.

here is what I did:
            #region 1. Get Register Address as Raw Byte value e.g. 0x12345678
            string raw = "0x12345678";
            #endregion

            #region 2. Decode it to Long as Big Endian
            //Raw to long
            long longValue = Convert.ToInt64(raw, 16);
            //Long to bytes (Littel endian)
            byte[] convertedLittelEndianBytes = BitConverter.GetBytes(longValue);
            //Bytes littel endian to bytes big endian
            byte[] convertedBigEndianBytes = new byte[convertedLittelEndianBytes.Length];
            convertedLittelEndianBytes.CopyTo(convertedBigEndianBytes, 0);
            Array.Reverse(convertedBigEndianBytes);
            //Bytes to string
            string convertedBigEndianString = BitConverter.ToString(convertedBigEndianBytes).Replace("-", string.Empty);
            //String to long
            long convertedLongBigEndian = long.Parse(convertedBigEndianString, System.Globalization.NumberStyles.HexNumber);
            #endregion

            #region 3. Use BitConverter for the converted Long value
            //Long to bytes
            byte[] convertedLongBytes = BitConverter.GetBytes(convertedLongBigEndian);
            #endregion

            #region 4. Compare the Raw Converted bytes with the Address you started from in step 1
            string convertedLongString = "0x" + BitConverter.ToString(convertedLongBytes).Replace("-", string.Empty);
            Debug.WriteLine("Raw: " + raw);
            Debug.WriteLine("Converted: " + convertedLongString);
            Assert.Equal(raw, convertedLongString);
            #endregion

**Comment by Touseefelahi** [comment]
@FaisalFagihi there would be one change in the 4th step above, as: compare with the bit converted data of step 1

**Comment by Touseefelahi** [comment]
it will require 4 bytes shift in case of gige vision layer as we discussed before

**Comment by Touseefelahi** [comment]
> @FaisalFagihi there would be one change in the 4th step above, as: compare with the bit converted data of step 1

@FaisalFagihi this is not correct, I mean the test is correct, the implementation is not correct

**Comment by Touseefelahi** [comment]
@FaisalFagihi Explanation: The goal is to use the write method as it is which takes two input values as:
`WriteMethod(long address, int value)

{
// Use bitconverter for address directly without the need of Array.Reverse method for GVCP big endiance requirement
}`

Let me know if there is any confusion about the task
